api/src/main/java/com/valtech/aem/saas/api/resource/PathTransformer.java

```diff
@@ -10,24 +10,42 @@
 
   /**
    * Gets one or more externalized links for the passed path argument. It retrieves more than one item in case an
-   * internal path can be represented by more than 1 url. (e.g. for indexing)
+   * internal path can be represented by more than one url. (e.g. for indexing). Links are appended with the html
+   * extension
    *
-   * @param request      sling request used for context.
-   * @param resourcePath resource/page path to be externalized.
+   * @param request sling request used for context.
+   * @param path    resource/page path to be externalized.
    * @return list of externalized paths
    */
-  List<String> externalize(SlingHttpServletRequest request, String resourcePath);
+  List<String> externalizeList(SlingHttpServletRequest request, String path);
 
-  default List<String> externalize(String resourcePath) {
-    return externalize(null, resourcePath);
+  /**
+   * Gets one or more externalized links for the passed path argument. It retrieves more than one item in case an
+   * internal path can be represented by more than one url. (e.g. for indexing) Links are appended with the html
+   * extension
+   *
+   * @param path resource/page path to be externalized.
+   * @return list of externalized paths
+   */
+  default List<String> externalizeList(String path) {
+    return externalizeList(null, path);
   }
 
   /**
-   * Retrieves a path that is product of the applied mapping rules.
+   * Gets an externalized link for the passed path argument. Link is appended with the html extension
+   *
+   * @param request sling request used for context.
+   * @param path    resource/page path to be externalized.
+   * @return externalized path
+   */
+  String externalize(SlingHttpServletRequest request, String path);
+
+  /**
+   * Retrieves a path that is a result of the applied mapping rules.
```

Comment:
Please mention here as well that by default the implementation uses resourceResolver.map . If customization is required a higher serviceranking than X should be used

Reply:
according http://docs.osgi.org/javadoc/r2/org/osgi/framework/Constants.html#SERVICE_RANKING, default value is 0